### PR TITLE
LibraryPanels: Fix issue where viewer with folder edit permissions could not update library panel

### LINKED ISSUE #56997: Edit permissions not applying for Library Panels (folderId not sent in the API request)
**What happened**:

A "Viewer" in an instance that has been given "Editor" permissions within a folder is able to create library panels within the folder, but cannot edit or save changes to library panels.

This occurs both when the "Viewer" user is granted Edit permissions to a folder directly or are a member of a Group with folder Edit permissions.

A similar issue is reported here: https://github.com/grafana/support-escalations/issues/2487 This issue was marked resolved, but this specific case was not solved.

https://user-images.githubusercontent.com/5699976/195871764-01ac5d6c-075f-4861-a345-86a873fc12a8.mov

_First half shows permissions as seen by my Admin user.
Second half (after opening other window) shows behavior when editing and creating library panels with user that should have correct privileges via Folder permissions._

**What you expected to happen**:
There are problems present in your service that are more likely causes, the mere omission of folder ID seems to imply a move to the root/general folder. 

**How to reproduce it (as minimally and precisely as possible)**:
@sakjur shared steps to reproduce that behavior based on main/48c27872af9ef without running the enterprise extensions.

---

Create the folder

```
curl 'http://admin:admin@localhost:3000/api/folders' -H 'Content-Type: application/json' -d '{"title":"Folder", "uid": "9jsYZNS4z"}'
```

Create the viewer user

```
curl 'https://admin:admin@localhost:3000/api/admin/users' \
-H 'Content-Type: application/json' \
--data-binary '{"name":"viewer","email":"","login":"viewer","password":"viewer"}'
```

extract the ID for the newly created user and add it as an editor to the folder.

```
curl 'http://admin:admin@localhost:3000/api/folders/9jsYZNS4z/users/$USER_ID' -H 'Content-Type: application/json' -d '{"permission":"Edit"}'
```

create the library element

```
curl 'http://viewer:viewer@localhost:3000/api/library-elements' \
-H 'Content-Type: application/json' \
--data-binary '{"folderId":1,"name":"Panel Title","model":{"libraryPanel":{"name":"Panel Title"},"id":2,"gridPos":{"x":0,"y":0,"w":12,"h":9},"type":"timeseries","title":"Panel Title","options":{"tooltip":{"mode":"single","sort":"none"},"legend":{"showLegend":true,"displayMode":"list","placement":"bottom","calcs":[]}},"fieldConfig":{"defaults":{"custom":{"drawStyle":"line","lineInterpolation":"linear","barAlignment":0,"lineWidth":1,"fillOpacity":0,"gradientMode":"none","spanNulls":false,"showPoints":"auto","pointSize":5,"stacking":{"mode":"none","group":"A"},"axisPlacement":"auto","axisLabel":"","axisColorMode":"text","scaleDistribution":{"type":"linear"},"axisCenteredZero":false,"hideFrom":{"tooltip":false,"viz":false,"legend":false},"thresholdsStyle":{"mode":"off"}},"color":{"mode":"palette-classic"},"mappings":[],"thresholds":{"mode":"absolute","steps":[{"value":null,"color":"green"},{"value":80,"color":"red"}]}},"overrides":[]}},"kind":1}'
```

Extract the UID for the library element and update it using the example from https://grafana.com/docs/grafana/latest/developers/http_api/library_element/
This fails because it assumes that you want to move the library element to 

```
curl "http://viewer:viewer@localhost:3000/api/library-elements/$PANEL_UID" \
-X 'PATCH' \
-H 'Content-Type: application/json' \
-d '{
  "name": "Renamed library panel",
  "kind": 1,
  "version": 1
}'
```

Add _folderId_ and you can now save.

```
curl "http://viewer:viewer@localhost:3000/api/library-elements/$PANEL_UID" \
-X 'PATCH' \
-H 'Content-Type: application/json' \
-d '{
  "name": "Renamed library panel",
  "kind": 1,
  "version": 1,
  "folderId": 8
}'
```

It seems like folderId is omitted from the request that the frontend sends when updating a library element.

```
{
    "name": "Panel Title",
    "model": {
        "libraryPanel": {
            "description": "",
            "meta": {
                "connectedDashboards": 1,
                "created": "2022-10-12T05:44:47Z",
                "createdBy": {
                    "avatarUrl": "/avatar/46d229b033af06a191ff2267bca9ae56",
                    "id": 1,
                    "name": "admin"
                },
                "folderName": "Folder",
                "folderUid": "9jsYZNS4z",
                "updated": "2022-10-12T05:49:56Z",
                "updatedBy": {
                    "avatarUrl": "/avatar/46d229b033af06a191ff2267bca9ae56",
                    "id": 1,
                    "name": "admin"
                }
            },
            "name": "Panel Title",
            "type": "timeseries",
            "uid": "DpqlWHS4z",
            "version": 2
        },
        "id": 2,
        "gridPos": {
            "h": 8,
            "w": 12,
            "x": 0,
            "y": 0
        },
        "type": "timeseries",
        "title": "Panel Title",
        "datasource": {
            "type": "loki",
            "uid": "A57Wu8n4z"
        },
        "description": "",
        "fieldConfig": {
            "defaults": {
                "custom": {
                    "drawStyle": "line",
                    "lineInterpolation": "linear",
                    "barAlignment": 0,
                    "lineWidth": 1,
                    "fillOpacity": 0,
                    "gradientMode": "none",
                    "spanNulls": false,
                    "showPoints": "auto",
                    "pointSize": 5,
                    "stacking": {
                        "mode": "none",
                        "group": "A"
                    },
                    "axisPlacement": "auto",
                    "axisLabel": "",
                    "axisColorMode": "text",
                    "scaleDistribution": {
                        "type": "linear"
                    },
                    "axisCenteredZero": false,
                    "hideFrom": {
                        "tooltip": false,
                        "viz": false,
                        "legend": false
                    },
                    "thresholdsStyle": {
                        "mode": "off"
                    }
                },
                "color": {
                    "mode": "palette-classic"
                },
                "mappings": [],
                "thresholds": {
                    "mode": "absolute",
                    "steps": [
                        {
                            "color": "green",
                            "value": null
                        },
                        {
                            "color": "red",
                            "value": 80
                        }
                    ]
                }
            },
            "overrides": []
        },
        "options": {
            "tooltip": {
                "mode": "single",
                "sort": "none"
            },
            "legend": {
                "showLegend": true,
                "displayMode": "list",
                "placement": "bottom",
                "calcs": []
            }
        }
    },
    "version": 2,
    "kind": 1
}
```

**Environment**:
- Grafana version: I followed the repro steps from this issue and was able to reproduce this issue in Enterprise 9.1.7 (on-prem) and in Grafana Cloud on 9.1.6
- User OS & Browser: Chrome

### PR DESCRIPTION
**Pull request by kaydelaney**
Fixes an issue where a viewer with folder edit permissions could not update a library panel.

**Which issue(s) does this PR fix?**:

Closes #56997

### CODE REVIEW

File: public/app/features/library-panels/state/api.ts [hunk 1]
```diff
@@ -90,9 +90,10 @@ export async function addLibraryPanel(
 
 export async function updateLibraryPanel(panelSaveModel: PanelModelWithLibraryPanel): Promise<LibraryElementDTO> {
   const { libraryPanel, ...model } = panelSaveModel;
-  const { uid, name, version } = libraryPanel;
+  const { uid, name, version, folderId } = libraryPanel;
   const kind = LibraryElementKind.Panel;
   const { result } = await getBackendSrv().patch(`/api/library-elements/${uid}`, {
+    folderId,
```
Comment (axelavargas) [inline public/app/features/library-panels/state/api.ts:96]: 🤔 we have [this other PR](https://github.com/grafana/grafana/pull/58393/files#diff-f3bc123a92bfa4aa9371dc4b06800bff62598140f08a8c5538f3c3bff627f7abR13) who is deprecating the usage of `folderId` and is starting to use `folderUid`, is it possible to use that here? or should we wait until that PR is merged?

Comment (kaydelaney) [reply public/app/features/library-panels/state/api.ts:96]: Might be worth waiting until that PR is merged

**Review by ivanortegaalba — APPROVED** [review]
Now, it works perfectly well! 💯 

|Before|After|
|-|-|
|

**Comment by axelavargas** [comment]
🤔 I don't think we can put this fix in 9.2.7, the dependant PR  https://github.com/grafana/grafana/pull/58393 is targeting 9.4 , I am removing the milestone and backport cc. @grafana/dashboards-squad

**Pull request merged**